### PR TITLE
Helloworld

### PR DESCRIPTION
**Pull request by aethos**
there are so many issues

### CODE REVIEW

File: src/ast.ml [hunk 1]
```diff
@@ -7,20 +7,32 @@ type bin_op =
...
+  | Set_t of types
+  | Map_t of types * types
+  | Tuple_t of types list
+
+(*
 type p_type = Int_t | Bool_t | Float_t | Char_t | Unit_t
 
 type cont_type =
-  String_t
+    String_t
```
Comment (Howon) [inline src/ast.ml:28]: lets use the old approach

File: src/codegen.ml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+(* Code generation: translate takes a semantically checked AST and
+produces LLVM IR
+
+LLVM tutorial: Make sure to read the OCaml version of the tutorial
```
Comment (vlad-scherbich) [inline src/codegen.ml]: Let's remove this tutorial piece?

File: src/codegen.ml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+http://llvm.org/docs/tutorial/index.html
+
+Detailed documentation on the OCaml LLVM library:
```
Comment (vlad-scherbich) [inline src/codegen.ml]: And this.

File: test_hello_world.osc [hunk 4]
```diff
@@ -0,0 +1,3 @@
+def main() => unit = {
+    print("Hello World!");
+}
```
Comment (vlad-scherbich) [inline test_hello_world.osc]: add a newline please

Comment (aethos) [reply test_hello_world.osc]: newline where

File: src/codegen.ml (hunk 2 above)
Comment (aethos) [reply src/codegen.ml]: sure

File: src/codegen.ml (hunk 3 above)
Comment (aethos) [reply src/codegen.ml]: sure

File: src/ast.ml (hunk 1 above)
Comment (aethos) [reply src/ast.ml:28]: the old approach doesn't actually work

File: test_hello_world.osc (hunk 4 above)
Comment (vlad-scherbich) [reply test_hello_world.osc]: after '}'

File: src/parser.mly [hunk 5]
```diff
@@ -146,12 +146,15 @@ pattern_list:
...
 
+/* TODO: Mut and Vassign do not exist in the AST! */
 mut_vdecl:
-  MUTABLE typ ID { Mut($3, $2) }
-  | MUTABLE typ ID ASSIGN expr { Mut($3, $2); Assign($2, $5) }
+/* nothing */ { Continue }
+ /* MUTABLE typ ID { Mut($3, $2) }
+  | MUTABLE typ ID ASSIGN expr { Mut($3, $2); Assign($2, $5) } */
 
+/* TODO: this needs and action, which will go in the AST! */
```
Comment (vlad-scherbich) [inline src/parser.mly]: `and` or `an`?

Comment (aethos) [reply src/parser.mly]: vlad you suck

File: src/ast.ml [hunk 6]
```diff
@@ -45,7 +57,7 @@ type actor_type = Actor_t of string
 
 type pool_type = Pool_t of actor_type * int
 
-and expr =
+type expr =
     Binop of expr * bin_op * expr
```
Comment (Howon) [inline src/ast.ml:61]: indentation

File: src/ast.ml [hunk 7]
```diff
@@ -56,51 +68,51 @@ and expr =
...
-  | Tuple_Lit of Tuple_t * expr list
+  | Tuple_Lit of Tuple_t * expr list *)
   | Call of string * expr list
   | Actor_comm of message * actor_op * actor_type
   | Noexpr
 
-and stmt =
-    Block of stt list
+type stmt =
+    Block of stmt list
```
Comment (Howon) [inline src/ast.ml:80]: comment

File: src/ast.ml [hunk 8]
```diff
@@ -56,51 +68,51 @@ and expr =
...
-  | If of expr * stmt * stmt
-  | For of int * int * int * stmt
-  | While of expr * stmt
+  | If of expr * stmt * stmt list
+  | For of string * int * int * int * stmt list
+  | While of expr * stmt list
   | Break
   | Continue
 
 type pattern = {
```
Comment (Howon) [inline src/ast.ml:91]: indentation

File: src/scanner.mll [hunk 9]
```diff
@@ -47,12 +47,12 @@ rule token = parse
   | '&'                   { BITWISE_AND }
   | '|'                   { BITWISE_OR }
   | '^'                   { BITWISE_XOR }
-  | ">>"                  { BITSHIFT_RIGHT }
-  | "<<"                  { BITSHIFT_LEFT }
+  | ">>"                  { BITWISE_RIGHT }
```
Comment (vlad-scherbich) [inline src/scanner.mll:50]: +1 for `BITSHIFT_` --> `BITWISE_`

File: hello_world.txt [hunk 10]
```diff
@@ -0,0 +1,41 @@
+-----hello world-----
```
Comment (Howon) [inline hello_world.txt]: put this somewhere else

File: src/codegen.ml [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+http://llvm.moe/ocaml/
+
+*)
+
+module L = Llvm
+module A = Ast
+
+module StringMap = Map.Make(String)
+
+let translate (messages, actors, functions) =
```
Comment (vlad-scherbich) [inline src/codegen.ml:6]: +1 on indentation

**Review by vlad-scherbich — APPROVED** [review]

File: src/ast.ml [hunk 12]
```diff
@@ -35,8 +47,8 @@ type optional = Optional_t of optional_rec | None_t*)
 type formal = string * types
 
 type message = {
-    name: string;
-    formals: formal list;
+    m_name: string;
```
Comment (Howon) [inline src/ast.ml:50]: indentation

File: src/codegen.ml [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+  let context = L.global_context () in
```
Comment (Howon) [inline src/codegen.ml:7]: make this uniform

**Review by Howon — APPROVED** [review]

**Pull request merged**